std/datetime/date.d

```diff
@@ -6594,12 +6594,12 @@ public:
             $(REF DateTimeException,std,datetime,date) if the given day is an
             invalid day of the year.
       +/
-    @property void dayOfYear(int day) @safe pure
+    @property void dayOfYear(int day) @safe pure nothrow @nogc
     {
         immutable int[] lastDay = isLeapYear ? lastDayLeap : lastDayNonLeap;
 
         if (day <= 0 || day > (isLeapYear ? daysInLeapYear : daysInYear))
-            throw new DateTimeException("Invalid day of the year.");
+            assert(0, "Invalid day of the year.");
```

Comment:
I suggest this be moved anyway outside this PR because it changes semantics, whereas everything else doesn't. For the matter in question, it seems excessive to assert(0) on wrong user input. We don't do that in most places.